diff --git a/src/llama-chat.cpp b/src/llama-chat.cpp
index dd27a381..47550954 100644
--- a/src/llama-chat.cpp
+++ b/src/llama-chat.cpp
@@ -6,11 +6,7 @@
 #include <sstream>
 #include <algorithm>
 
-#if __cplusplus >= 202000L
-    #define LU8(x) (const char*)(u8##x)
-#else
-    #define LU8(x) u8##x
-#endif
+#define LU8(x) reinterpret_cast<const char*>(u8##x)
 
 // trim whitespace from the beginning and end of a string
 static std::string trim(const std::string & str) {
