diff --git a/ggml/src/ggml-metal/ggml-metal.m b/ggml/src/ggml-metal/ggml-metal.m
index 456e1fd9..fb84a4e4 100644
--- a/ggml/src/ggml-metal/ggml-metal.m
+++ b/ggml/src/ggml-metal/ggml-metal.m
@@ -72,7 +72,9 @@ static id<MTLDevice> ggml_backend_metal_device_acq(struct ggml_backend_metal_dev
     assert(ctx != NULL);
 
     if (ctx->mtl_device == nil) {
-        ctx->mtl_device = MTLCreateSystemDefaultDevice();
+        NSArray<id<MTLDevice>> *devices = MTLCopyAllDevices();
+        assert(devices.count != 0);
+        ctx->mtl_device = devices.firstObject;
     }
 
     if (ctx->mtl_device) {
