diff --git a/include/grpc/event_engine/memory_request.h b/include/grpc/event_engine/memory_request.h
index 5c74bdb..abc1234 100644
--- a/include/grpc/event_engine/memory_request.h
+++ b/include/grpc/event_engine/memory_request.h
@@ -17,6 +17,7 @@
 #include <grpc/support/port_platform.h>
 #include <stddef.h>
 
+#include <string>
 #include "absl/strings/string_view.h"
 
 namespace grpc_event_engine {
