diff -rupN a/lib/tls/openssl/openssl-server.c b/lib/tls/openssl/openssl-server.c
--- libwebsockets-4.3.2/lib/tls/openssl/openssl-server.c	2022-05-17 18:53:27.000000000 +0200
+++ libwebsockets-4.3.2-changed/lib/tls/openssl/openssl-server.c	2023-07-24 15:50:08.529365302 +0200
@@ -709,7 +709,7 @@ lws_tls_server_new_nonblocking(struct lw
 	return 0;
 }

-int
+enum lws_ssl_capable_status
 lws_tls_server_abort_connection(struct lws *wsi)
 {
 	if (wsi->tls.use_ssl)
