diff --git a/include/bustache/model.hpp b/include/bustache/model.hpp
index 575969a..07bc89d 100644
--- a/include/bustache/model.hpp
+++ b/include/bustache/model.hpp
@@ -330,6 +330,13 @@ namespace bustache::detail
             buf[count++] = c;
         }
 
+        void append(const char* const begin, const char* const end)
+        {
+            for (const char* it = begin; it != end; ++it) {
+                push_back(*it);
+            }
+        }
+
         void flush() { os(buf, count); }
 
         std::size_t count = 0;
