diff --git a/google/cloud/storage/google_cloud_cpp_storage.cmake b/google/cloud/storage/google_cloud_cpp_storage.cmake
index 5e53e972a9..7c1f1b6542 100644
--- a/google/cloud/storage/google_cloud_cpp_storage.cmake
+++ b/google/cloud/storage/google_cloud_cpp_storage.cmake
@@ -399,6 +399,21 @@ install(
     DESTINATION include/google/cloud/storage/testing
     COMPONENT google_cloud_cpp_development)
 
+add_library(
+    mock_google_cloud_client # cmake-format: sort
+    testing/mock_client.cc
+    testing/mock_client.h)
+target_link_libraries(
+    mock_google_cloud_client
+    PUBLIC  google-cloud-cpp::storage
+            GTest::gmock)
+google_cloud_cpp_add_common_options(mock_google_cloud_client)
+target_include_directories(
+    mock_google_cloud_client PUBLIC $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}>
+                                    $<INSTALL_INTERFACE:include>)
+target_compile_options(mock_google_cloud_client
+                        PUBLIC ${GOOGLE_CLOUD_CPP_EXCEPTIONS_FLAG})
+
 if (BUILD_TESTING)
     add_library(
         storage_client_testing # cmake-format: sort
