diff --color=auto -rupN llama.cpp-b8944/CMakeLists.txt llama.cpp-b8944-patched/CMakeLists.txt
--- llama.cpp-b8944/CMakeLists.txt	2026-04-27 08:30:55
+++ llama.cpp-b8944-patched/CMakeLists.txt	2026-04-27 13:49:25
@@ -191,6 +191,7 @@ add_subdirectory(src)
 #
 
 add_subdirectory(src)
+add_subdirectory(tools/mtmd)
 
 #
 # utils, programs, examples and tests
diff --color=auto -rupN llama.cpp-b8944/common/ngram-mod.cpp llama.cpp-b8944-patched/common/ngram-mod.cpp
--- llama.cpp-b8944/common/ngram-mod.cpp	2026-04-27 08:30:55
+++ llama.cpp-b8944-patched/common/ngram-mod.cpp	2026-04-30 08:28:08
@@ -1,4 +1,5 @@
 #include "ngram-mod.h"
+#include <algorithm>
 
 //
 // common_ngram_mod
diff --color=auto -rupN llama.cpp-b8944/tools/mtmd/CMakeLists.txt llama.cpp-b8944-patched/tools/mtmd/CMakeLists.txt
--- llama.cpp-b8944/tools/mtmd/CMakeLists.txt	2026-04-27 08:30:55
+++ llama.cpp-b8944-patched/tools/mtmd/CMakeLists.txt	2026-04-27 13:50:45
@@ -101,20 +101,6 @@ endif()
     endif()
 endif()
 
-add_executable(llama-llava-cli    deprecation-warning.cpp)
-add_executable(llama-gemma3-cli   deprecation-warning.cpp)
-add_executable(llama-minicpmv-cli deprecation-warning.cpp)
-add_executable(llama-qwen2vl-cli  deprecation-warning.cpp)
-
-set(TARGET llama-mtmd-cli)
-add_executable         (${TARGET} mtmd-cli.cpp)
-set_target_properties  (${TARGET} PROPERTIES OUTPUT_NAME llama-mtmd-cli)
-if(LLAMA_TOOLS_INSTALL)
-    install(TARGETS ${TARGET} RUNTIME)
-endif()
-target_link_libraries  (${TARGET} PRIVATE llama-common mtmd Threads::Threads)
-target_compile_features(${TARGET} PRIVATE cxx_std_17)
-
 # mtmd-debug tool
 add_executable(llama-mtmd-debug debug/mtmd-debug.cpp)
 set_target_properties(llama-mtmd-debug PROPERTIES OUTPUT_NAME llama-mtmd-debug)
