diff -rupN a/sdk/core/CMakeLists.txt b/sdk/core/CMakeLists.txt
--- a/sdk/core/CMakeLists.txt	2023-07-11 04:18:12.000000000 +0200
+++ b/sdk/core/CMakeLists.txt	2023-07-25 16:58:17.919348377 +0200
@@ -10,11 +10,6 @@ set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)

 add_subdirectory(azure-core)

-if (NOT BUILD_WINDOWS_UWP)
-  message(STATUS "Including AMQP library")
-  add_subdirectory(azure-core-amqp)
-endif()
-
 if (NOT DISABLE_AZURE_CORE_OPENTELEMETRY)
   add_subdirectory(azure-core-tracing-opentelemetry)
 endif()
