diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6b51ac64..8720c375 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -39,7 +39,7 @@
 cmake_minimum_required(VERSION 3.7)
 message(STATUS "Using CMake version ${CMAKE_VERSION}")
 
-set(CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake" ${CMAKE_MODULE_PATH})
+list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
 
 include(CheckFunctionExists)
 include(CheckSymbolExists)
