diff --git a/crt/aws-crt-cpp/include/aws/crt/StringView.h b/crt/aws-crt-cpp/include/aws/crt/StringView.h
index a0fec83..6d0b997 100644
--- a/crt/aws-crt-cpp/include/aws/crt/StringView.h
+++ b/crt/aws-crt-cpp/include/aws/crt/StringView.h
@@ -13,6 +13,7 @@
 #include <iterator>
 #include <limits>
 #include <stddef.h>
+#include <string>
 #include <type_traits>
 
 #if __cplusplus >= 201703L || (defined(_MSC_LANG) && _MSC_LANG >= 201703L)
