diff --git a/generated/src/aws-cpp-sdk-kinesis/CMakeLists.txt b/generated/src/aws-cpp-sdk-kinesis/CMakeLists.txt
index 099657cffe0..0b8c7a01a99 100644
--- a/generated/src/aws-cpp-sdk-kinesis/CMakeLists.txt
+++ b/generated/src/aws-cpp-sdk-kinesis/CMakeLists.txt
@@ -59,6 +59,11 @@ if(USE_WINDOWS_DLL_SEMANTICS AND BUILD_SHARED_LIBS)
     target_compile_definitions(${PROJECT_NAME} PRIVATE "AWS_KINESIS_EXPORTS")
 endif()
 
+if(FORCE_EXPORT_KINESIS_API)
+    target_compile_definitions(${PROJECT_NAME} PUBLIC "AWS_KINESIS_API=__declspec(dllexport)")
+    target_compile_definitions(${PROJECT_NAME} PUBLIC "AWS_KINESIS_EXTERN=")
+endif()
+
 target_include_directories(${PROJECT_NAME} PUBLIC
     $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
     $<INSTALL_INTERFACE:include>)
diff --git a/generated/src/aws-cpp-sdk-kinesis/include/aws/kinesis/Kinesis_EXPORTS.h b/generated/src/aws-cpp-sdk-kinesis/include/aws/kinesis/Kinesis_EXPORTS.h
index 237bbdfcc8c..20065a815f9 100644
--- a/generated/src/aws-cpp-sdk-kinesis/include/aws/kinesis/Kinesis_EXPORTS.h
+++ b/generated/src/aws-cpp-sdk-kinesis/include/aws/kinesis/Kinesis_EXPORTS.h
@@ -22,10 +22,13 @@
 #define AWS_KINESIS_API __declspec(dllimport)
 #endif /* AWS_KINESIS_EXPORTS */
 #define AWS_KINESIS_EXTERN
-#else
-#define AWS_KINESIS_API
-#define AWS_KINESIS_EXTERN extern
 #endif  // USE_IMPORT_EXPORT
+#ifndef AWS_KINESIS_API
+#define AWS_KINESIS_API
+#endif
+#ifndef AWS_KINESIS_EXTERN
+#define AWS_KINESIS_EXTERN
+#endif
 #else   // defined (USE_WINDOWS_DLL_SEMANTICS) || defined (WIN32)
 #define AWS_KINESIS_API
 #define AWS_KINESIS_EXTERN extern
diff --git a/generated/src/aws-cpp-sdk-s3/CMakeLists.txt b/generated/src/aws-cpp-sdk-s3/CMakeLists.txt
index f44ba566427..8f3d11734b5 100644
--- a/generated/src/aws-cpp-sdk-s3/CMakeLists.txt
+++ b/generated/src/aws-cpp-sdk-s3/CMakeLists.txt
@@ -59,6 +59,11 @@ if(USE_WINDOWS_DLL_SEMANTICS AND BUILD_SHARED_LIBS)
     target_compile_definitions(${PROJECT_NAME} PRIVATE "AWS_S3_EXPORTS")
 endif()
 
+if(FORCE_EXPORT_S3_API)
+    target_compile_definitions(${PROJECT_NAME} PUBLIC "AWS_S3_API=__declspec(dllexport)")
+    target_compile_definitions(${PROJECT_NAME} PUBLIC "AWS_S3_EXTERN=")
+endif()
+
 target_include_directories(${PROJECT_NAME} PUBLIC
     $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
     $<INSTALL_INTERFACE:include>)
diff --git a/generated/src/aws-cpp-sdk-s3/include/aws/s3/S3_EXPORTS.h b/generated/src/aws-cpp-sdk-s3/include/aws/s3/S3_EXPORTS.h
index e1a1c22b2da..e6a727abc9c 100644
--- a/generated/src/aws-cpp-sdk-s3/include/aws/s3/S3_EXPORTS.h
+++ b/generated/src/aws-cpp-sdk-s3/include/aws/s3/S3_EXPORTS.h
@@ -22,10 +22,13 @@
 #define AWS_S3_API __declspec(dllimport)
 #endif /* AWS_S3_EXPORTS */
 #define AWS_S3_EXTERN
-#else
-#define AWS_S3_API
-#define AWS_S3_EXTERN extern
 #endif  // USE_IMPORT_EXPORT
+#ifndef AWS_S3_API
+#define AWS_S3_API
+#endif
+#ifndef AWS_S3_EXTERN
+#define AWS_S3_EXTERN
+#endif
 #else   // defined (USE_WINDOWS_DLL_SEMANTICS) || defined (WIN32)
 #define AWS_S3_API
 #define AWS_S3_EXTERN extern
diff --git a/src/aws-cpp-sdk-core/CMakeLists.txt b/src/aws-cpp-sdk-core/CMakeLists.txt
index c6a0f1c262e..5e21622a80e 100644
--- a/src/aws-cpp-sdk-core/CMakeLists.txt
+++ b/src/aws-cpp-sdk-core/CMakeLists.txt
@@ -634,6 +634,11 @@ elseif (BUILD_SHARED_LIBS)
     target_compile_definitions(${PROJECT_NAME} PRIVATE "SMITHY_EXPORTS=1")
 endif()
 
+if(FORCE_EXPORT_CORE_API)
+    target_compile_definitions(${PROJECT_NAME} PUBLIC "AWS_CORE_API=__declspec(dllexport)")
+    target_compile_definitions(${PROJECT_NAME} PUBLIC "AWS_CORE_EXTERN=")
+endif()
+
 set_compiler_flags(${PROJECT_NAME})
 set_compiler_warnings(${PROJECT_NAME})
 
diff --git a/src/aws-cpp-sdk-core/include/aws/core/Core_EXPORTS.h b/src/aws-cpp-sdk-core/include/aws/core/Core_EXPORTS.h
index ecd5a39db7b..93f8887aff0 100644
--- a/src/aws-cpp-sdk-core/include/aws/core/Core_EXPORTS.h
+++ b/src/aws-cpp-sdk-core/include/aws/core/Core_EXPORTS.h
@@ -19,11 +19,14 @@
             #define  AWS_CORE_API __declspec(dllimport)
         #endif // AWS_CORE_EXPORTS
         #define AWS_CORE_EXTERN
-    #else // USE_IMPORT_EXPORT
-        #define AWS_CORE_API
-        #define AWS_CORE_EXTERN extern
     #endif // USE_IMPORT_EXPORT
     #define AWS_CORE_LOCAL
+    #ifndef AWS_CORE_API
+        #define AWS_CORE_API
+    #endif
+    #ifndef AWS_CORE_EXTERN
+        #define AWS_CORE_EXTERN
+    #endif
 #else // defined (USE_WINDOWS_DLL_SEMANTICS) || defined (_WIN32)
     #define AWS_CORE_API
     #define AWS_CORE_EXTERN extern
diff --git a/src/aws-cpp-sdk-core/include/aws/core/endpoint/DefaultEndpointProvider.h b/src/aws-cpp-sdk-core/include/aws/core/endpoint/DefaultEndpointProvider.h
index e9be6700828..8b656c70676 100644
--- a/src/aws-cpp-sdk-core/include/aws/core/endpoint/DefaultEndpointProvider.h
+++ b/src/aws-cpp-sdk-core/include/aws/core/endpoint/DefaultEndpointProvider.h
@@ -6,6 +6,7 @@
 
 #pragma once
 
+#include <aws/core/Core_EXPORTS.h>
 #include <aws/core/endpoint/AWSPartitions.h>
 #include <aws/core/endpoint/EndpointProviderBase.h>
 #include <aws/core/endpoint/EndpointParameter.h>
diff --git a/src/aws-cpp-sdk-core/include/aws/core/endpoint/EndpointProviderBase.h b/src/aws-cpp-sdk-core/include/aws/core/endpoint/EndpointProviderBase.h
index b4781f6e5c7..c4c4d580fba 100644
--- a/src/aws-cpp-sdk-core/include/aws/core/endpoint/EndpointProviderBase.h
+++ b/src/aws-cpp-sdk-core/include/aws/core/endpoint/EndpointProviderBase.h
@@ -6,6 +6,7 @@
 
 #pragma once
 
+#include <aws/core/Core_EXPORTS.h>
 #include <aws/core/endpoint/AWSEndpoint.h>
 #include <aws/core/client/AWSError.h>
 #include <aws/core/endpoint/EndpointParameter.h>
diff --git a/generated/src/aws-cpp-sdk-s3-crt/include/aws/s3-crt/S3Crt_EXPORTS.h b/generated/src/aws-cpp-sdk-s3-crt/include/aws/s3-crt/S3Crt_EXPORTS.h
index 65af31b5521..de9ec13724d 100644
--- a/generated/src/aws-cpp-sdk-s3-crt/include/aws/s3-crt/S3Crt_EXPORTS.h
+++ b/generated/src/aws-cpp-sdk-s3-crt/include/aws/s3-crt/S3Crt_EXPORTS.h
@@ -22,10 +22,13 @@
 #define AWS_S3CRT_API __declspec(dllimport)
 #endif /* AWS_S3CRT_EXPORTS */
 #define AWS_S3CRT_EXTERN
-#else
-#define AWS_S3CRT_API
-#define AWS_S3CRT_EXTERN extern
 #endif  // USE_IMPORT_EXPORT
+#ifndef AWS_S3CRT_API
+#define AWS_S3CRT_API
+#endif
+#ifndef AWS_S3CRT_EXTERN
+#define AWS_S3CRT_EXTERN
+#endif
 #else   // defined (USE_WINDOWS_DLL_SEMANTICS) || defined (WIN32)
 #define AWS_S3CRT_API
 #define AWS_S3CRT_EXTERN extern
diff --git a/generated/src/aws-cpp-sdk-s3-crt/CMakeLists.txt b/generated/src/aws-cpp-sdk-s3-crt/CMakeLists.txt
index 946962d93fe..48159eda753 100644
--- a/generated/src/aws-cpp-sdk-s3-crt/CMakeLists.txt
+++ b/generated/src/aws-cpp-sdk-s3-crt/CMakeLists.txt
@@ -59,6 +59,11 @@ if(USE_WINDOWS_DLL_SEMANTICS AND BUILD_SHARED_LIBS)
     target_compile_definitions(${PROJECT_NAME} PRIVATE "AWS_S3CRT_EXPORTS")
 endif()

+if(FORCE_EXPORT_S3_CRT_API)
+    target_compile_definitions(${PROJECT_NAME} PUBLIC "AWS_S3CRT_API=__declspec(dllexport)")
+    target_compile_definitions(${PROJECT_NAME} PUBLIC "AWS_S3CRT_EXTERN=")
+endif()
+
 target_include_directories(${PROJECT_NAME} PUBLIC
     $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
     $<INSTALL_INTERFACE:include>)
