diff --git a/crt/aws-crt-cpp/crt/s2n/CMakeLists.txt b/crt/aws-crt-cpp/crt/s2n/CMakeLists.txt
index f5ee8379..eb3b2591 100644
--- a/crt/aws-crt-cpp/crt/s2n/CMakeLists.txt
+++ b/crt/aws-crt-cpp/crt/s2n/CMakeLists.txt
@@ -402,17 +402,17 @@ if (LIBCRYPTO_SUPPORTS_EVP_MD5_SHA1_HASH)
     target_compile_options(${PROJECT_NAME} PUBLIC -DS2N_LIBCRYPTO_SUPPORTS_EVP_MD5_SHA1_HASH)
 endif()
 
-# Determine if RC4 is available in libcrypto
-try_compile(
-        LIBCRYPTO_SUPPORTS_EVP_RC4
-        ${CMAKE_BINARY_DIR}
-        SOURCES "${CMAKE_CURRENT_LIST_DIR}/tests/features/evp_rc4.c"
-        LINK_LIBRARIES ${LINK_LIB} ${OS_LIBS}
-)
-
-if (LIBCRYPTO_SUPPORTS_EVP_RC4)
-    target_compile_options(${PROJECT_NAME} PUBLIC -DS2N_LIBCRYPTO_SUPPORTS_EVP_RC4)
-endif()
+# # Determine if RC4 is available in libcrypto
+# try_compile(
+#         LIBCRYPTO_SUPPORTS_EVP_RC4
+#         ${CMAKE_BINARY_DIR}
+#         SOURCES "${CMAKE_CURRENT_LIST_DIR}/tests/features/evp_rc4.c"
+#         LINK_LIBRARIES ${LINK_LIB} ${OS_LIBS}
+# )
+
+# if (LIBCRYPTO_SUPPORTS_EVP_RC4)
+#     target_compile_options(${PROJECT_NAME} PUBLIC -DS2N_LIBCRYPTO_SUPPORTS_EVP_RC4)
+# endif()
 
 # Determine if EVP_MD_CTX_set_pkey_ctx is available in libcrypto
 try_compile(
diff --git a/crt/aws-crt-cpp/crt/s2n/s2n.mk b/crt/aws-crt-cpp/crt/s2n/s2n.mk
index 3b0e1500..acf87e15 100644
--- a/crt/aws-crt-cpp/crt/s2n/s2n.mk
+++ b/crt/aws-crt-cpp/crt/s2n/s2n.mk
@@ -217,11 +217,11 @@ ifeq ($(TRY_EVP_MD5_SHA1_HASH), 0)
 	DEFAULT_CFLAGS += -DS2N_LIBCRYPTO_SUPPORTS_EVP_MD5_SHA1_HASH
 endif
 
-# Determine if EVP_md5_sha1 is available
-TRY_EVP_RC4 := $(call try_compile,$(S2N_ROOT)/tests/features/evp_rc4.c)
-ifeq ($(TRY_EVP_RC4), 0)
-	DEFAULT_CFLAGS += -DS2N_LIBCRYPTO_SUPPORTS_EVP_RC4
-endif
+# # Determine if EVP_md5_sha1 is available
+# TRY_EVP_RC4 := $(call try_compile,$(S2N_ROOT)/tests/features/evp_rc4.c)
+# ifeq ($(TRY_EVP_RC4), 0)
+# 	DEFAULT_CFLAGS += -DS2N_LIBCRYPTO_SUPPORTS_EVP_RC4
+# endif
 
 # Determine if EVP_MD_CTX_set_pkey_ctx is available
 TRY_EVP_MD_CTX_SET_PKEY_CTX := $(call try_compile,$(S2N_ROOT)/tests/features/evp_md_ctx_set_pkey_ctx.c)
