diff --git a/crt/aws-crt-cpp/crt/aws-c-io/source/event_loop.c b/crt/aws-crt-cpp/crt/aws-c-io/source/event_loop.c
index c37aaf6..2a45c21 100644
--- a/crt/aws-crt-cpp/crt/aws-c-io/source/event_loop.c
+++ b/crt/aws-crt-cpp/crt/aws-c-io/source/event_loop.c
@@ -114,7 +114,7 @@ static struct aws_event_loop_group *s_event_loop_group_new(
 
     el_group->allocator = alloc;
     aws_ref_count_init(
-        &el_group->ref_count, el_group, (aws_simple_completion_callback *)s_aws_event_loop_group_shutdown_async);
+        &el_group->ref_count, el_group, (aws_simple_completion_callback *)s_aws_event_loop_group_shutdown_sync);
 
     if (aws_array_list_init_dynamic(&el_group->event_loops, alloc, el_count, sizeof(struct aws_event_loop *))) {
         goto on_error;
