Subject: [PATCH] remove FindLZ4 installation
---
Index: CMakeLists.txt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	(revision 2af6749287b3695edf2421631ebf08d6bcb176e3)
+++ b/CMakeLists.txt	(date 1751541934552)
@@ -262,10 +262,7 @@
     COMPATIBILITY AnyNewerVersion
 )

-install(
-    FILES "${project_config}" "${project_version}" "packaging/cmake/Modules/FindLZ4.cmake"
-    DESTINATION "${config_install_dir}"
-)
+#A KURVA ANYÁD

 install(
     EXPORT "${targets_export_name}"
