diff -rupN orig/CMakeLists.txt patched/CMakeLists.txt
--- orig/CMakeLists.txt	2019-11-09 20:08:01.000000000 +0100
+++ patched/CMakeLists.txt	2019-12-17 15:22:40.000000000 +0100
@@ -29,8 +29,7 @@ option(SOCI_ASAN "Enable address sanitiz
 ###############################################################################
 
 # Path to additional CMake modules
-set(CMAKE_MODULE_PATH ${SOCI_SOURCE_DIR}/cmake ${CMAKE_MODULE_PATH})
-set(CMAKE_MODULE_PATH ${SOCI_SOURCE_DIR}/cmake/modules ${CMAKE_MODULE_PATH})
+list(APPEND CMAKE_MODULE_PATH "${SOCI_SOURCE_DIR}/cmake/modules" "${SOCI_SOURCE_DIR}/cmake")
 
 include(SociUtilities)
 include(SociConfig)
@@ -204,4 +203,3 @@ endforeach()
 configure_file("${CONFIG_FILE_IN}" "${CONFIG_FILE_OUT}")
 
 message(STATUS "")
-
