diff -ur a/CMake/AbseilDll.cmake b/CMake/AbseilDll.cmake
--- a/CMake/AbseilDll.cmake	2026-08-18 14:53:30.000000000 +0200
+++ b/CMake/AbseilDll.cmake	2026-09-03 13:22:10.161368576 +0200
@@ -548,7 +548,7 @@
   "crc_cord_state"
   "crc_cpu_detect"
   "crc_internal"
-  "crc32c"
+  "crc32c_internal"
   "debugging"
   "debugging_internal"
   "demangle_internal"
diff -ur a/absl/crc/BUILD.bazel b/absl/crc/BUILD.bazel
--- a/absl/crc/BUILD.bazel	2026-08-18 14:53:30.000000000 +0200
+++ b/absl/crc/BUILD.bazel	2026-09-03 13:22:10.161668055 +0200
@@ -60,7 +60,7 @@
 )
 
 cc_library(
-    name = "crc32c",
+    name = "crc32c_internal",
     srcs = [
         "crc32c.cc",
         "internal/crc32c_inline.h",
diff -ur a/absl/crc/CMakeLists.txt b/absl/crc/CMakeLists.txt
--- a/absl/crc/CMakeLists.txt	2026-08-18 14:53:30.000000000 +0200
+++ b/absl/crc/CMakeLists.txt	2026-09-03 13:22:10.161889990 +0200
@@ -38,7 +38,7 @@
 
 absl_cc_library(
   NAME
-    crc32c
+  crc32c_internal
   HDRS
     "crc32c.h"
     "internal/crc32c.h"
@@ -71,7 +71,7 @@
   COPTS
     ${ABSL_DEFAULT_COPTS}
   DEPS
-    absl::crc32c
+    absl::crc32c_internal
     absl::strings
     absl::str_format
     GTest::gtest_main
@@ -111,7 +111,7 @@
   COPTS
     ${ABSL_DEFAULT_COPTS}
   DEPS
-    absl::crc32c
+    absl::crc32c_internal
     absl::memory
     absl::random_random
     absl::random_distributions
@@ -141,7 +141,7 @@
   COPTS
     ${ABSL_DEFAULT_COPTS}
   DEPS
-    absl::crc32c
+    absl::crc32c_internal
     absl::config
     absl::strings
     absl::no_destructor
@@ -156,6 +156,6 @@
     ${ABSL_DEFAULT_COPTS}
   DEPS
     absl::crc_cord_state
-    absl::crc32c
+    absl::crc32c_internal
     GTest::gtest_main
 )
diff -ur a/absl/strings/CMakeLists.txt b/absl/strings/CMakeLists.txt
--- a/absl/strings/CMakeLists.txt	2026-08-18 14:53:30.000000000 +0200
+++ b/absl/strings/CMakeLists.txt	2026-09-03 13:22:10.162082289 +0200
@@ -1075,7 +1075,7 @@
     absl::cordz_update_scope
     absl::cordz_update_tracker
     absl::core_headers
-    absl::crc32c
+    absl::crc32c_internal
     absl::crc_cord_state
     absl::endian
     absl::function_ref
