diff --git a/CMake/AbseilDll.cmake b/CMake/AbseilDll.cmake
index 32cc28fb..8e9dc702 100644
--- a/CMake/AbseilDll.cmake
+++ b/CMake/AbseilDll.cmake
@@ -521,7 +521,7 @@ set(ABSL_INTERNAL_DLL_TARGETS
   "crc_cord_state"
   "crc_cpu_detect"
   "crc_internal"
-  "crc32c"
+  "crc32c_internal"
   "debugging"
   "debugging_internal"
   "demangle_internal"
diff --git a/absl/crc/BUILD.bazel b/absl/crc/BUILD.bazel
index 890d637c..8d680c22 100644
--- a/absl/crc/BUILD.bazel
+++ b/absl/crc/BUILD.bazel
@@ -72,7 +72,7 @@ cc_library(
 )
 
 cc_library(
-    name = "crc32c",
+    name = "crc32c_internal",
     srcs = [
         "crc32c.cc",
         "internal/crc32c_inline.h",
diff --git a/absl/crc/CMakeLists.txt b/absl/crc/CMakeLists.txt
index d52a1bc4..3cec9dcb 100644
--- a/absl/crc/CMakeLists.txt
+++ b/absl/crc/CMakeLists.txt
@@ -53,7 +53,7 @@ absl_cc_library(
 
 absl_cc_library(
   NAME
-    crc32c
+  crc32c_internal
   HDRS
     "crc32c.h"
     "internal/crc32c.h"
@@ -86,7 +86,7 @@ absl_cc_test(
   COPTS
     ${ABSL_DEFAULT_COPTS}
   DEPS
-    absl::crc32c
+    absl::crc32c_internal
     absl::strings
     absl::str_format
     GTest::gtest_main
@@ -126,7 +126,7 @@ absl_cc_test(
   COPTS
     ${ABSL_DEFAULT_COPTS}
   DEPS
-    absl::crc32c
+    absl::crc32c_internal
     absl::memory
     absl::random_random
     absl::random_distributions
@@ -156,7 +156,7 @@ absl_cc_library(
   COPTS
     ${ABSL_DEFAULT_COPTS}
   DEPS
-    absl::crc32c
+    absl::crc32c_internal
     absl::config
     absl::strings
     absl::no_destructor
@@ -171,6 +171,6 @@ absl_cc_test(
     ${ABSL_DEFAULT_COPTS}
   DEPS
     absl::crc_cord_state
-    absl::crc32c
+    absl::crc32c_internal
     GTest::gtest_main
 )
diff --git a/absl/strings/CMakeLists.txt b/absl/strings/CMakeLists.txt
index 3a1619e8..2c368cf2 100644
--- a/absl/strings/CMakeLists.txt
+++ b/absl/strings/CMakeLists.txt
@@ -987,7 +987,7 @@ absl_cc_library(
     absl::cordz_update_scope
     absl::cordz_update_tracker
     absl::core_headers
-    absl::crc32c
+    absl::crc32c_internal
     absl::crc_cord_state
     absl::endian
     absl::function_ref
