diff --git a/include/meta/meta.hpp b/include/meta/meta.hpp
index 1372703e7..66ee2928b 100644
--- a/include/meta/meta.hpp
+++ b/include/meta/meta.hpp
@@ -3776,7 +3776,7 @@ namespace meta
         /// \ingroup integral
         template <char... Chs>
         constexpr fold<list<char_<Chs>...>, meta::size_t<0>, quote<detail::atoi_>>
-            operator"" _z()
+            operator""_z()
         {
             return {};
         }
diff --git a/include/range/v3/utility/compressed_pair.hpp b/include/range/v3/utility/compressed_pair.hpp
index 438d12f0d..a9590b4bd 100644
--- a/include/range/v3/utility/compressed_pair.hpp
+++ b/include/range/v3/utility/compressed_pair.hpp
@@ -92,8 +92,7 @@ namespace ranges
         };
 
         template<typename... Ts>
-        using compressed_tuple RANGES_DEPRECATED(
-            "ranges::compressed_tuple is deprecated.") =
+        using compressed_tuple =
             compressed_tuple_<meta::list<Ts...>,
                               meta::make_index_sequence<sizeof...(Ts)>>;
     } // namespace compressed_tuple_detail
