diff --git a/cmake/ThirdPartyDependencies.cmake b/cmake/ThirdPartyDependencies.cmake
index f02af02..1a51f82 100644
--- a/cmake/ThirdPartyDependencies.cmake
+++ b/cmake/ThirdPartyDependencies.cmake
@@ -15,48 +15,48 @@ function(declare_system_library target)
                                              $<TARGET_PROPERTY:${target},INTERFACE_INCLUDE_DIRECTORIES>)
 endfunction()
 
-if(NOT TARGET fmt::fmt)
-  # https://github.com/fmtlib/fmt/releases
-  cpmaddpackage(
-    NAME
-    fmt
-    GIT_TAG
-    11.0.1
-    VERSION
-    11.0.1
-    GITHUB_REPOSITORY
-    "fmtlib/fmt"
-    EXCLUDE_FROM_ALL ON
-    OPTIONS
-    "FMT_INSTALL OFF"
-    # Unicode support for MSVC enabled in CompilerWarnings.cmake
-    "FMT_UNICODE OFF"
-    "FMT_DOC OFF"
-    "BUILD_SHARED_LIBS OFF"
-    "CMAKE_C_VISIBILITY_PRESET hidden"
-    "CMAKE_CXX_VISIBILITY_PRESET hidden"
-    "CMAKE_POSITION_INDEPENDENT_CODE ON")
-endif()
-
-if(NOT TARGET spdlog::spdlog)
-  # https://github.com/gabime/spdlog/releases
-  cpmaddpackage(
-    NAME
-    spdlog
-    VERSION
-    1.14.1
-    GITHUB_REPOSITORY
-    "gabime/spdlog"
-    EXCLUDE_FROM_ALL ON
-    OPTIONS
-    "SPDLOG_INSTALL OFF"
-    "BUILD_SHARED_LIBS OFF"
-    "CMAKE_C_VISIBILITY_PRESET hidden"
-    "CMAKE_CXX_VISIBILITY_PRESET hidden"
-    "CMAKE_POSITION_INDEPENDENT_CODE ON"
-    "SPDLOG_BUILD_SHARED OFF"
-    "SPDLOG_FMT_EXTERNAL ON")
-endif()
+# if(NOT TARGET fmt::fmt)
+#   # https://github.com/fmtlib/fmt/releases
+#   cpmaddpackage(
+#     NAME
+#     fmt
+#     GIT_TAG
+#     11.0.1
+#     VERSION
+#     11.0.1
+#     GITHUB_REPOSITORY
+#     "fmtlib/fmt"
+#     EXCLUDE_FROM_ALL ON
+#     OPTIONS
+#     "FMT_INSTALL OFF"
+#     # Unicode support for MSVC enabled in CompilerWarnings.cmake
+#     "FMT_UNICODE OFF"
+#     "FMT_DOC OFF"
+#     "BUILD_SHARED_LIBS OFF"
+#     "CMAKE_C_VISIBILITY_PRESET hidden"
+#     "CMAKE_CXX_VISIBILITY_PRESET hidden"
+#     "CMAKE_POSITION_INDEPENDENT_CODE ON")
+# endif()
+
+# if(NOT TARGET spdlog::spdlog)
+#   # https://github.com/gabime/spdlog/releases
+#   cpmaddpackage(
+#     NAME
+#     spdlog
+#     VERSION
+#     1.14.1
+#     GITHUB_REPOSITORY
+#     "gabime/spdlog"
+#     EXCLUDE_FROM_ALL ON
+#     OPTIONS
+#     "SPDLOG_INSTALL OFF"
+#     "BUILD_SHARED_LIBS OFF"
+#     "CMAKE_C_VISIBILITY_PRESET hidden"
+#     "CMAKE_CXX_VISIBILITY_PRESET hidden"
+#     "CMAKE_POSITION_INDEPENDENT_CODE ON"
+#     "SPDLOG_BUILD_SHARED OFF"
+#     "SPDLOG_FMT_EXTERNAL ON")
+# endif()
 
 if(NOT TARGET Microsoft.GSL::GSL)
   # https://github.com/microsoft/GSL/releases
@@ -172,70 +172,70 @@ if(NOT TARGET taocpp::json)
 endif()
 
 
-if(NOT TARGET asio::asio)
-  # https://github.com/chriskohlhoff/asio/tags
-  cpmaddpackage(
-    NAME
-    asio
-    GIT_TAG
-    asio-1-31-0
-    VERSION
-    1.31.0
-    GITHUB_REPOSITORY
-    "chriskohlhoff/asio"
-    EXCLUDE_FROM_ALL ON)
-endif()
-
-# ASIO doesn't use CMake, we have to configure it manually. Extra notes for using on Windows:
-#
-# 1) If _WIN32_WINNT is not set, ASIO assumes _WIN32_WINNT=0x0501, i.e. Windows XP target, which is definitely not the
-# platform which most users target.
-#
-# 2) WIN32_LEAN_AND_MEAN is defined to make Winsock2 work.
-if(asio_ADDED)
-  add_library(asio STATIC ${asio_SOURCE_DIR}/asio/src/asio.cpp ${asio_SOURCE_DIR}/asio/src/asio_ssl.cpp)
-
-  target_include_directories(asio SYSTEM PUBLIC ${asio_SOURCE_DIR}/asio/include)
-  target_compile_definitions(asio PRIVATE ASIO_STANDALONE=1 ASIO_NO_DEPRECATED=1 ASIO_SEPARATE_COMPILATION=1)
-  target_link_libraries(asio PRIVATE Threads::Threads OpenSSL::SSL OpenSSL::Crypto)
-  set_target_properties(
-    asio
-    PROPERTIES C_VISIBILITY_PRESET hidden
-               CXX_VISIBILITY_PRESET hidden
-               POSITION_INDEPENDENT_CODE TRUE)
-
-  if(WIN32)
-    # macro see @ https://stackoverflow.com/a/40217291/1746503
-    macro(get_win32_winnt version)
-      if(CMAKE_SYSTEM_VERSION)
-        set(ver ${CMAKE_SYSTEM_VERSION})
-        string(REGEX MATCH "^([0-9]+).([0-9])" ver ${ver})
-        string(REGEX MATCH "^([0-9]+)" verMajor ${ver})
-        # Check for Windows 10, b/c we'll need to convert to hex 'A'.
-        if("${verMajor}" MATCHES "10")
-          set(verMajor "A")
-          string(REGEX REPLACE "^([0-9]+)" ${verMajor} ver ${ver})
-        endif("${verMajor}" MATCHES "10")
-        # Remove all remaining '.' characters.
-        string(REPLACE "." "" ver ${ver})
-        # Prepend each digit with a zero.
-        string(REGEX REPLACE "([0-9A-Z])" "0\\1" ver ${ver})
-        set(${version} "0x${ver}")
-      endif()
-    endmacro()
-
-    if(NOT DEFINED _WIN32_WINNT)
-      get_win32_winnt(ver)
-      set(_WIN32_WINNT ${ver})
-    endif()
-
-    message(STATUS "Set _WIN32_WINNT=${_WIN32_WINNT}")
-
-    target_compile_definitions(asio INTERFACE _WIN32_WINNT=${_WIN32_WINNT} WIN32_LEAN_AND_MEAN)
-  endif()
-
-  add_library(asio::asio ALIAS asio)
-endif()
+# if(NOT TARGET asio::asio)
+#   # https://github.com/chriskohlhoff/asio/tags
+#   cpmaddpackage(
+#     NAME
+#     asio
+#     GIT_TAG
+#     asio-1-31-0
+#     VERSION
+#     1.31.0
+#     GITHUB_REPOSITORY
+#     "chriskohlhoff/asio"
+#     EXCLUDE_FROM_ALL ON)
+# endif()
+
+# # ASIO doesn't use CMake, we have to configure it manually. Extra notes for using on Windows:
+# #
+# # 1) If _WIN32_WINNT is not set, ASIO assumes _WIN32_WINNT=0x0501, i.e. Windows XP target, which is definitely not the
+# # platform which most users target.
+# #
+# # 2) WIN32_LEAN_AND_MEAN is defined to make Winsock2 work.
+# if(asio_ADDED)
+#   add_library(asio STATIC ${asio_SOURCE_DIR}/asio/src/asio.cpp ${asio_SOURCE_DIR}/asio/src/asio_ssl.cpp)
+
+#   target_include_directories(asio SYSTEM PUBLIC ${asio_SOURCE_DIR}/asio/include)
+#   target_compile_definitions(asio PRIVATE ASIO_STANDALONE=1 ASIO_NO_DEPRECATED=1 ASIO_SEPARATE_COMPILATION=1)
+#   target_link_libraries(asio PRIVATE Threads::Threads OpenSSL::SSL OpenSSL::Crypto)
+#   set_target_properties(
+#     asio
+#     PROPERTIES C_VISIBILITY_PRESET hidden
+#                CXX_VISIBILITY_PRESET hidden
+#                POSITION_INDEPENDENT_CODE TRUE)
+
+#   if(WIN32)
+#     # macro see @ https://stackoverflow.com/a/40217291/1746503
+#     macro(get_win32_winnt version)
+#       if(CMAKE_SYSTEM_VERSION)
+#         set(ver ${CMAKE_SYSTEM_VERSION})
+#         string(REGEX MATCH "^([0-9]+).([0-9])" ver ${ver})
+#         string(REGEX MATCH "^([0-9]+)" verMajor ${ver})
+#         # Check for Windows 10, b/c we'll need to convert to hex 'A'.
+#         if("${verMajor}" MATCHES "10")
+#           set(verMajor "A")
+#           string(REGEX REPLACE "^([0-9]+)" ${verMajor} ver ${ver})
+#         endif("${verMajor}" MATCHES "10")
+#         # Remove all remaining '.' characters.
+#         string(REPLACE "." "" ver ${ver})
+#         # Prepend each digit with a zero.
+#         string(REGEX REPLACE "([0-9A-Z])" "0\\1" ver ${ver})
+#         set(${version} "0x${ver}")
+#       endif()
+#     endmacro()
+
+#     if(NOT DEFINED _WIN32_WINNT)
+#       get_win32_winnt(ver)
+#       set(_WIN32_WINNT ${ver})
+#     endif()
+
+#     message(STATUS "Set _WIN32_WINNT=${_WIN32_WINNT}")
+
+#     target_compile_definitions(asio INTERFACE _WIN32_WINNT=${_WIN32_WINNT} WIN32_LEAN_AND_MEAN)
+#   endif()
+
+#   add_library(asio::asio ALIAS asio)
+# endif()
 
 add_library(jsonsl OBJECT ${PROJECT_SOURCE_DIR}/third_party/jsonsl/jsonsl.c)
 set_target_properties(jsonsl PROPERTIES C_VISIBILITY_PRESET hidden POSITION_INDEPENDENT_CODE TRUE)
@@ -245,7 +245,7 @@ declare_system_library(snappy)
 declare_system_library(llhttp::llhttp)
 declare_system_library(hdr_histogram_static)
 declare_system_library(Microsoft.GSL::GSL)
-declare_system_library(spdlog::spdlog)
-declare_system_library(fmt::fmt)
-declare_system_library(asio)
+# declare_system_library(spdlog::spdlog)
+# declare_system_library(fmt::fmt)
+# declare_system_library(asio)
 declare_system_library(taocpp::json)
