diff --git a/CMakeLists.txt b/CMakeLists.txt
index 671fb2c..48273f5 100644
--- a/crt/aws-crt-cpp/crt/aws-c-io/CMakeLists.txt
+++ b/crt/aws-crt-cpp/crt/aws-c-io/CMakeLists.txt
@@ -141,18 +141,9 @@ if (USE_S2N)
         file(GLOB AWS_IO_TLS_SRC
                 "source/s2n/*.c"
                 )
-        # Prefer find_package() because it's the normal CMake way to do dependencies.
-        # But fall back on aws_use_package() because some projects still need to do an IN_SOURCE_BUILD of S2N.
-        # (e.g. aws-crt-java until this is resolved: https://github.com/awslabs/aws-crt-java/pull/817)
-        find_package(s2n QUIET)
-
-        if (s2n_FOUND)
-            list(APPEND DEP_AWS_LIBS AWS::s2n)
-        else()
-            # Set flag to use in-source path to  <s2n/unstable/*.h> headers if we do an IN_SOURCE_BUILD.
-            aws_use_package(s2n)
-            add_definitions(-DAWS_S2N_INSOURCE_PATH)
-        endif()
+        # Set flag to use in-source path to  <s2n/unstable/*.h> headers if we do an IN_SOURCE_BUILD.
+        aws_use_package(s2n)
+        add_definitions(-DAWS_S2N_INSOURCE_PATH)
 endif()
 
 file(GLOB IO_HEADERS
