diff --git a/cmake/modules/FindSQLite3.cmake b/cmake/modules/FindSQLite3.cmake
--- a/cmake/modules/FindSQLite3.cmake
+++ b/cmake/modules/FindSQLite3.cmake
@@ -17,6 +17,8 @@ find_path(SQLITE3_INCLUDE_DIR
 find_path(SQLITE3_INCLUDE_DIR
   NAMES sqlite3.h
   PATH_PREFIXES sqlite sqlite3
+  HINTS
+  $ENV{SQLITE_ROOT}/include
   PATHS
   /usr/include
   /usr/local/include
@@ -27,13 +29,14 @@ find_path(SQLITE3_INCLUDE_DIR
   $ENV{ProgramFiles}/SQLite3/*/include
   $ENV{SystemDrive}/SQLite/*/include
   $ENV{SystemDrive}/SQLite3/*/include
-  $ENV{SQLITE_ROOT}/include
   ${SQLITE_ROOT_DIR}/include
   $ENV{OSGEO4W_ROOT}/include)
 
 set(SQLITE3_NAMES sqlite3_i sqlite3)
 find_library(SQLITE3_LIBRARY
   NAMES ${SQLITE3_NAMES}
+  HINTS
+  $ENV{SQLITE_ROOT}/lib
   PATHS
   /usr/lib
   /usr/local/lib
@@ -42,7 +45,6 @@ find_library(SQLITE3_LIBRARY
   $ENV{ProgramFiles}/SQLite3/*/lib
   $ENV{SystemDrive}/SQLite/*/lib
   $ENV{SystemDrive}/SQLite3/*/lib
-  $ENV{SQLITE_ROOT}/lib
   ${SQLITE_ROOT_DIR}/lib
   $ENV{OSGEO4W_ROOT}/lib)
 
